bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/transport/HubIOStream.java

```diff
@@ -265,54 +272,30 @@ private boolean isClearedBuffer(String data) {
     /**
      * Helper method to fetch url from http server
      *
-     * @param resource the url
+     * @param path the url path
      * @return contents returned by http server
      * @throws IOException
      */
-    private String getURL(String resource) throws IOException {
-        String url = "http://" + host + ":" + port + resource;
+    private String getURL(String path) throws IOException {
+        Request request = httpClient.newRequest(host, port).path(path).header(HttpHeader.AUTHORIZATION, "Basic " + auth)
+                .timeout(30, TimeUnit.SECONDS);
+        logger.trace("getting {}", request.getURI());
 
-        HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
         try {
-            connection.setConnectTimeout(30000);
-            connection.setReadTimeout(10000);
-            connection.setUseCaches(false);
-            connection.setDoInput(true);
-            connection.setDoOutput(false);
-            connection.setRequestProperty("Authorization", "Basic " + auth);
-
-            logger.trace("getting {}", url);
-
-            int responseCode = connection.getResponseCode();
-            if (responseCode != 200) {
-                if (responseCode == 401) {
+            ContentResponse response = request.send();
+
+            int statusCode = response.getStatus();
+            switch (statusCode) {
+                case HttpStatus.OK_200:
+                    return response.getContentAsString();
+                case HttpStatus.UNAUTHORIZED_401:
                     throw new IOException(
                             "Bad username or password. See the label on the bottom of the hub for the correct login information.");
-                } else {
-                    throw new IOException(url + " failed with the response code: " + responseCode);
-                }
-            }
-
-            return getData(connection.getInputStream());
-        } finally {
-            connection.disconnect();
-        }
-    }
-
-    private String getData(InputStream is) throws IOException {
-        BufferedInputStream bis = new BufferedInputStream(is);
-        try {
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            byte[] buffer = new byte[1024];
-            int length = 0;
-            while ((length = bis.read(buffer)) != -1) {
-                baos.write(buffer, 0, length);
+                default:
+                    throw new IOException("GET " + request.getURI() + " failed with status code: " + statusCode);
             }
-
-            String s = baos.toString();
-            return s;
-        } finally {
-            bis.close();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
For `InterruptedException`, restore the interrupted state of the thread by calling `Thread.currentThread().interrupt()`, see https://rules.sonarsource.com/java/RSPEC-2142/. You could also let the exception propagate up the call stack to make sure the binding won't react to the `IOException` with any retry mechanism or similar when requested to stop.